### ISSUE TITLE
Keep consolidated cache inside a peer

### ISSUE THREAD

**Issue by Dilshat** [subutai-io/gorjun#149]
Currently each RH downloads templates separately even within the same peer.
Instead we need to somehow reuse the MH cache on all the connected RHs to facilitate environment creation

**Comment by akarasulu** [comment]
I've lost track of this. Maybe you can assist and correct me @Dilshat:

1. At one point each peer had a Kurjun instance running inside it
2. We then switched to Gorjun
3. We then made each RH have Gorjun
4. We then switched to using NGINX to cache instead of having Gorjun in each RH

Please correct me:

> Now we have each RH with an NGINX cache?

Thanks in advance!

**Comment by Dilshat** [comment]
In the current implementation we have a Gorjun per RH and an Nginx per RH.
However each RH has own nginx cache. So inside a peer, we have don't have one cache, we have several caches.

**Comment by Dilshat** [comment]
Moreover, there is one nuance in the logic of local Gorjun:
Local Gorjun now serves only the Console's requests: list templates, etc. 
When "subutai import {template}" command is executed, agent downloads templates from CDN directly (if not present locally) without using local Gorjun

**Comment by akarasulu** [comment]
Ok thanks. There are lots of problems here. I think the Gorjun team needs to think about all these deficiencies and perhaps we can attack them in the next few sprints. The good thing is there are **LOTS** of room for us to majorly improve performance with some minor Gorjun enhancements.

**Comment by Dilshat** [comment]
Yeah also need to check if local nginx caches the CDN artifacts at all.
